src/bip32/child_number.rs

```diff
@@ -0,0 +1,122 @@
+use std::{fmt, str::FromStr};
+
+const INDEX_THRESHOLD: u32 = 2147483648;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ChildNumber {
```

Comment:
what do you think giving a more explicit name to this structure? Something like DerivationStep, PathNode, KeyIndex or KeyStep?